src/server/pipeline.py

```diff
@@ -232,9 +244,30 @@ async def perform_handshake(
                 f" established"
             )
         )
+        # Disconnect feeler connections and update connection timestamp.
+        if connection.connection_type == NodeType.FULL_NODE:
+            if connection.is_outbound:
+                await global_connections.mark_good(connection.get_peer_info())
+                await global_connections.update_connection_time(
+                    connection.get_peer_info()
+                )
+                if connection.is_feeler:
+                    connection.close()
+                    global_connections.close(connection)
+                    return
+                # Request peers after handshake.
+                if connection.local_type == NodeType.FULL_NODE:
+                    await connection.send(Message("request_peers", ""))
+
         # Only yield a connection if the handshake is succesful and the connection is not a duplicate.
         yield connection, global_connections
     except Exception as e:
+        if (
+            connection.connection_type == NodeType.FULL_NODE
```

Comment:
The pipeline shouldn't have to know about node types. Maybe `_state_changed` can introduce a failed connection notification.

Reply:
Again, state changed is being used by the RPC server. 
Perhaps we can modify the state_changed code so that you can have multiple consumers of state changed (multiple callbacks). So it's not just the RPC server listening for state changed, but also some networking code. 

It is kind of confusing though, Since the interface between the networking layer (pipeline.py) and service layer (full_node.py) is the calling of API methods and yielding of responses. If we rely on state_changed instead to communicate between these layers, it could become messy.

Reply:
Ha, our crosstalk jinxed. I think this is a great idea, and although it does require understanding more stuff, "pub/sub" (for "publish/subscribe") is a really great model in fairly common usage, and can help a lot in removing circular dependencies (and circular dependencies are spaghetti because you can't understand anything in isolation).